unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendRequestHUD/SentFriendRequestHUDController.cs

```diff
@@ -86,6 +86,7 @@ private void Show(string friendRequestId)
             var ownProfile = userProfileBridge.GetOwn();
             view.SetSenderProfilePicture(ownProfile.snapshotObserver);
 
+            view.SetSortingOrder(dataStore.HUDs.currentPassportSortingOrder.Get() + 1);
```

Comment:
I think some logic of Sent and Recieve Controllers should be generalized to remove so much of duplicated code. Such practices as composition, generics, extracting common interface should be used (but avoiding inheritance)
![image](https://user-images.githubusercontent.com/35366872/211575851-f1b1298e-11af-4358-ac3c-b7994e28d66d.png)

Reply:
and this comment is not only about this function, but in general about both controllers. I am checking their `Hide`, `OpenProfile`, `ShowOrHide` methods... they are all the same. Remember the DRY meta-principle (but keep in mind AHA) 😉

Reply:
To be able to achieve this refactor, following composition, i would have to add an internal controller, which injects a new `IVIew`, since it would be wrong to use the interfaces of the received/sent. The solution gets more complex in my opinion, harder to follow and harder to customize if needed. The code is not 100% the same, it has small differences (which perhaps can be parametrized, im not sure). I think the effort does not worth the gain. If we encounter issues on maintaining both classes because of duplications, then we can think of proper refactor.

Reply:
I encounter such design with Places and Events UI, where adding one small thing was requiring to touch 6 classes (3 views + 3 controller) just because of such duplicated logic, that is why I don't want to have more of this. Also - the logic was same, but methods order wasn't, so it was very difficult to follow and even find similarities. (Refactoring of similar Controllers in Places and Events you can find [here ](https://github.com/decentraland/unity-renderer/pull/3917/files). I am not sure that it needs adding internal controller and new IView, but rather usage of Strategy pattern and some SRP extracted helper-classes to handle duplicated logic)

@mikhail-dcl, @AjimenezDCL, what would be your opinion on it? I will give a green and pass it to you, guys

Reply:
cc @Kinerius (as you like such discussions ;) )